pkg/util/ulid/ulid_test.go

```diff
@@ -0,0 +1,868 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// Copyright 2016 The Oklog Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code originated in github.com/oklog/ulid.
+
+package ulid_test
+
+import (
+	"bytes"
+	crand "crypto/rand"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"strings"
+	"testing"
+	"testing/iotest"
+	"testing/quick"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/ulid"
+	"github.com/cockroachdb/errors"
+)
+
+func ExampleULID() {
+	t := timeutil.Unix(1000000, 0)
+	entropy := ulid.Monotonic(rand.New(rand.NewSource(t.UnixNano())), 0)
+	fmt.Println(ulid.MustNew(ulid.Timestamp(t), entropy))
+	// Output: 0000XSNJG0MQJHBF4QX1EFD6Y3
+}
+
+func TestNew(t *testing.T) {
+	t.Parallel() // SAFE FOR TESTING (this comment is for the linter)
```

Comment:
I had to adapt both `ulid.go` and `ulid_test.go` to met linter rules, but here I cheated.

As `t.Parallel` is forbidden by the linter, I added  exceptions for them.

- Are you ok with this exceptions here? 
- Is best to just remove every `t.Parallel`? 
- Do I have to adapt the code to use `sync.WorkGroups` and rewrite all the tests?

Reply:
Personally I'm fine with this. Doing the least amount of work here makes sense, probably worse off to change the tests more.

Reply:
👌